application/listeners/listeners.go

```diff
@@ -128,3 +135,36 @@ func sendNewMessageNotification(e events.Event) {
 		}
 	}
 }
+
+func sendPostStatusUpdatedNotification(e events.Event) {
+	if e.Kind != domain.EventApiPostStatusUpdated {
+		return
+	}
+
+	pEData, ok := e.Payload["eventData"].(models.PostStatusEventData)
+	if !ok {
+		domain.ErrLogger.Print("unable to parse Post Status Updated event payload")
+		return
+	}
+
+	if pEData.Post.Type != models.PostTypeRequest {
+		return
+	}
+
+	switch pEData.NewStatus {
```

Comment:
Instead of the two levels of `switch`es, would it work to have a map of templates where the key is the two statuses concatenated?

Reply:
Sounds good.